SimpleCalculator/SimpleCalculatorTest/Test.cs

```diff
@@ -0,0 +1,86 @@
+using System.Reflection;
+using SimpleCalculator.Business;
+using SimpleCalculator.Business.Enums;
+using System;
+
+namespace SimpleCalculatorTest;
+
+public class Test
+{
+    [Theory]
+    [InlineData(2, 3)]
+    [InlineData(5, 6)]
+    [InlineData(13, 6)]
+    public void Calculate_Sum_SumOfTwoNumbers(int firstOperand, int secondOperand)
+    {
+        int expected = firstOperand + secondOperand;
+        Calculator calculator = new Calculator();
+
+        int answer = calculator.Calculate(firstOperand, secondOperand, OperatorEnum.sum);
+
+        Assert.True(expected == answer);
+    }
+    
+    [Theory]
+    [InlineData(2, 3)]
+    [InlineData(5, 6)]
+    [InlineData(13, 6)]
+    public void Calculate_Sub_SumOfTwoNumbers(int firstOperand, int secondOperand)
+    {
+        int expected = firstOperand - secondOperand;
+        Calculator calculator = new Calculator();
+
+        int answer = calculator.Calculate(firstOperand, secondOperand, OperatorEnum.sub);
+
+        Assert.True(expected == answer);
+    }
+
+    [Theory]
+    [InlineData(0, 1000)]
+    [InlineData(-100, 1000)]
+    [InlineData(2, 3)]
+    [InlineData(5, 6)]
+    [InlineData(13, 6)]
+    public void Calculate_Multiply_MultiplyOfTwoNumbers(int firstOperand, int secondOperand)
+    {
+        int expected = firstOperand * secondOperand;
+        Calculator calculator = new Calculator();
+
+        int answer = calculator.Calculate(firstOperand, secondOperand, OperatorEnum.multiply);
+
+        Assert.True(expected == answer);
+    }
+
+    [Theory]
+    [InlineData(0, 1000)]
+    [InlineData(-100, 1000)]
+    [InlineData(2, 3)]
+    [InlineData(5, 6)]
+    [InlineData(13, 6)]
+    public void Calculate_Division_DivisionOfTwoNumbers(int firstOperand, int secondOperand)
+    {
+        int expected = firstOperand / secondOperand;
+        Calculator calculator = new Calculator();
+
+        int answer = calculator.Calculate(firstOperand, secondOperand, OperatorEnum.division);
+
+        Assert.True(expected == answer);
+    }
+
+    [Fact]
+    public void Calculate_DivisionByZero_Exception() {
+        int firstOperand = 12;
+        int secondOperand = 0;
+        Calculator calculator = new Calculator();
+
+        try
+        {
+            int answer = calculator.Calculate(firstOperand, secondOperand, OperatorEnum.division);
+            Assert.True(false);
+        }
+        catch
+        {
+            Assert.True(true);
+        }
+    }
+}
```

Comment:
Only an exception and the 4 basic operations are tested, and in each test not only is the Calculator class being tested but OperatorProvidore as well, this violates the principles of Unit Testing, which states that we have to test individual units of source code.

Reply:
I don't understand why this is incorrect and how to correct this.

Reply:
You can mock IOperatorProvider.